source/rules/no-unsafe-switchmap.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { stripIndent } from "common-tags";
+import decamelize from "decamelize";
+import { Rule } from "eslint";
+import esquery from "esquery";
+import * as es from "estree";
+import { isCallExpression, isIdentifier, isLiteral, typecheck } from "../utils";
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description: "Disallows unsafe switchMap usage in Effects and epics.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Unsafe switchMap usage in Effects and epics is forbidden."
+    },
+    schema: [
+      {
+        properties: {
+          allow: {
+            oneOf: [
+              { type: "string" },
+              { type: "array", items: { type: "string" } }
+            ]
+          },
+          disallow: {
+            oneOf: [
+              { type: "string" },
+              { type: "array", items: { type: "string" } }
+            ]
+          },
+          observable: {
+            oneOf: [
+              { type: "string" },
+              { type: "array", items: { type: "string" } }
+            ]
+          }
+        },
+        type: "object",
+        description: stripIndent`
+          An optional object with optional \`allow\`, \`disallow\` and \`observable\` properties.
+          The properties can be specified as regular expression strings or as arrays of words.
+          The \`allow\` or \`disallow\` properties are mutually exclusive. Whether or not
+          \`switchMap\` is allowed will depend upon the matching of action types with \`allow\` or \`disallow\`.
+          The \`observable\` property is used to identify the action observables from which effects and epics are composed.
+        `
+      }
+    ]
+  },
+  create: context => {
+    const defaultObservable = "action(s|\\$)?";
+    const defaultDisallow = [
+      "add",
+      "create",
+      "delete",
+      "post",
+      "put",
+      "remove",
+      "set",
+      "update"
+    ];
+
+    function createRegExp(value: any): RegExp | null {
+      if (!value || !value.length) {
+        return null;
+      }
+      const flags = "i";
+      if (typeof value === "string") {
+        return new RegExp(value, flags);
+      }
+      const words = value as string[];
+      const joined = words.map(word => `(\\b|_)${word}(\\b|_)`).join("|");
+      return new RegExp(`(${joined})`, flags);
+    }
+
+    let allowRegExp: RegExp | null;
+    let disallowRegExp: RegExp | null;
+    let observableRegExp: RegExp;
+
+    const [config] = context.options;
+    if (config && (config.allow || config.disallow)) {
+      allowRegExp = createRegExp(config.allow);
+      disallowRegExp = createRegExp(config.disallow);
+      observableRegExp = new RegExp(config.observable || defaultObservable);
```

Comment:
I'm not sure if is already said, but this is the 2nd time I encounter this (and I believe I forgot to mention/take a deeper look into this before).
In TSLint the RegExp is made with the case invariant flag.
If we create the RegExp here with the same flag, we get an invalid ESLint query exception. 


```
SyntaxError: Syntax error in selector "CallExpression[callee.property.name='pipe'][callee.object.name=/action(s|\$)?/i]" at position 78: Expected
" " or "]" but "i" found.
```

Reply:
I think the best we can do is `[callee.object.name=/[Aa]ction(s|\$)?/]`.

Reply:
That seems to be the only option (if we want to use a esquery selector). 
The only thing I could find was the following issue https://github.com/estools/esquery/issues/65